### PR TITLE
add multiple method calls for unit test grader

### PR DESCRIPTION
**Pull request by mint87**

### CODE REVIEW

**Comment by mint87** [comment]
https://codio.myjetbrains.com/youtrack/issue/codio-11232

**Comment by mloginov** [comment]
need to support structures like this

```
def testOne(self):\n' +
                  '    self.assertEqual(maxindex([0, 2, 4]),2,' +
                          '"Calling function <code>maxindex([0, 2, 4])</code>.")\n' +
                  '    self.assertEqual(maxindex([7, 2, 4]),0,' +
                          '"Calling function <code>maxindex([7, 2, 4])</code>.")\n' +
                  '    self.assertEqual(maxindex([7, 8, 4]),1,' +
                          '"Calling function <code>maxindex([7, 8, 4])</code>.")\n' +
```

so interfaces should be like this

```
export interface AssertEqualParams {
  methodCall: string;
  expectedOutput: string;
  errorMessage?: string;
}

export interface UnitTest {
  testName: string, // maybe store test name (testOne for example)
  assertEquals: AssertEqualParams[]
}

```

**Comment by destitutus** [comment]
conflicts here

File: src/lib/render.ts [hunk 1]
```diff
@@ -256,7 +256,7 @@ const renderUnitTestCodePrepend = (code?: string): Cash => {
   return codePrependContainer
 }
 
-export const renderUnitTest = (test?: UnitTest | undefined): Cash => {
+export const renderUnitTest = (test?: AssertEqualParams): Cash => {
```
Comment (mloginov) [inline src/lib/render.ts]: `UnitTest` should be passed as param to `renderUnitTest`

File: src/lib/render.ts [hunk 2]
```diff
@@ -321,7 +323,7 @@ const renderUnitTestGrader = (options?: ParsonsOptions): Cash => {
   const testsList: Cash = $('<ul class="tests-list"></ul>')
 
   if (tests) {
-    tests.forEach((test: UnitTest) => testsList.append(renderUnitTest(test)))
+    tests.forEach((test: UnitTest) => testsList.append(renderUnitTest(test.assertEquals)))
```
Comment (mloginov) [inline src/lib/render.ts]: `test` should be passed as param to `renderUnitTest`

File: src/lib/data-helper.ts [hunk 3]
```diff
@@ -109,16 +110,32 @@ const collectVariableCheckGraderOptions = (container: Cash): VariableCheckGrader
   }
 }
 
-export const collectUnitTest = (container: Cash): UnitTest => {
+export const collectUnitTest = (container: Cash): AssertEqualParams => {
```
Comment (mloginov) [inline src/lib/data-helper.ts]: `collectUnitTest` should collect `UnitTest` not `AssertEqualParams`
with `UnitTest.name`

File: src/lib/data-helper.ts [hunk 4]
```diff
@@ -109,16 +110,32 @@ const collectVariableCheckGraderOptions = (container: Cash): VariableCheckGrader
...
+    obj.errorMessage = errorMessage[i] || ''
+    return obj
+  })
+  return methods.map((method: { [k: string]: string }) => ('    self.assertEqual('
+      + `${method.methodCall},${method.expectedOutput},${method.errorMessage})`)).join('\n')
+}
+
 const collectUnitTestGraderOptions = (container: Cash): UnitTestGraderOptions => {
-  const unitTests: UnitTest[] = []
+  const unitTests: AssertEqualParams[] = []
```
Comment (mloginov) [inline src/lib/data-helper.ts]: should be `const unitTests: UnitTest[] = []`

File: src/lib/data-helper.ts [hunk 5]
```diff
@@ -130,9 +147,9 @@ const collectUnitTestGraderOptions = (container: Cash): UnitTestGraderOptions =>
   const unitTestsArr: string[] = [
     'import unittestparson',
     'class myTests(unittestparson.unittest):',
-    ...unitTests.map((test: UnitTest, index: number): string => [
+    ...unitTests.map((test: AssertEqualParams, index: number): string => [
```
Comment (mloginov) [inline src/lib/data-helper.ts]: should be `test: UnitTest`

File: src/lib/data-helper.ts [hunk 6]
```diff
@@ -130,9 +147,9 @@ const collectUnitTestGraderOptions = (container: Cash): UnitTestGraderOptions =>
...
       `  def test_${index}(self):`,
```
Comment (mloginov) [inline src/lib/data-helper.ts]: need to restore name from `unitTest.name`

**Review by mloginov — CHANGES_REQUESTED** [review]

File: src/lib/data-helper.ts [hunk 7]
```diff
@@ -147,9 +151,9 @@ const collectUnitTestGraderOptions = (container: Cash): UnitTestGraderOptions =>
   const unitTestsArr: string[] = [
     'import unittestparson',
     'class myTests(unittestparson.unittest):',
-    ...unitTests.map((test: AssertEqualParams, index: number): string => [
-      `  def test_${index}(self):`,
-      `${getMethodsCall(test)}`
+    ...unitTests.map((test: UnitTest): string => [
+      `  def ${test.name}(self):`,
```
Comment (mloginov) [inline src/lib/data-helper.ts]: if `test.name` is empty need to use `test_${index}`

File: src/lib/render.ts [hunk 8]
```diff
@@ -256,8 +256,9 @@ const renderUnitTestCodePrepend = (code?: string): Cash => {
   return codePrependContainer
 }
 
-export const renderUnitTest = (test?: AssertEqualParams): Cash => {
+export const renderUnitTest = (test?: UnitTest | undefined): Cash => {
   const testContainer: Cash = $('<li class="test-container"></li>')
+  $(testContainer).data('data-name', test ? test.name : '')
```
Comment (mloginov) [inline src/lib/render.ts]: maybe change to `test-name`?

**Review by mloginov — CHANGES_REQUESTED** [review]

**Review by mloginov — APPROVED** [review]

**Pull request merged**